### ISSUE TITLE
Server shuts down immediately

### ISSUE THREAD

**Issue by broxen** [Segs/Segs#115]
After latest commits, server will not startup, and produces only the following nondescript error message:
```
main
Reading map data from ./data/bin/ folder
Shutting down AdminServer: "No reason"
Shutting down game server because :  "Full system shutdown"
MapServer has not been started yet.
```

I'm looking through the various ShutDown calls now.

### SEGS version
Clone latest from git

### Operating system
Linux (Fedora 26)

### C++ compiler and version
gcc (GCC) 7.2.1 20170915 (Red Hat 7.2.1-2)

**Comment by nemerle** [comment]
Likely one of the data files is missing but server does not report it

**Comment by nemerle** [comment]
```
segs v0.4.0 alpha (The Real Beginning)
main
Config file is missing 'location_addr' entry, will try to use default
Config file is missing 'auth_pass' entry, will try to use default
Reading map data from ./data/bin/ folder
Reading ./data/bin/costume.bin ... OK
Reading ./data/bin/supergroupColors.bin ... OK
Loading origins:
Reading ./data/bin/origins.bin ... OK
Reading ./data/bin/villain_origins.bin ... OK
Loading classes:
Reading ./data/bin/classes.bin ... OK
Reading ./data/bin/villain_classes.bin ... OK
 All map data read
 Postprocessing runtime data ..  Hashes filled
```
this is how it should look on success..
I'm investigating reporting code now.

**Comment by nemerle** [comment]
Please try the most recent commit ?

**Comment by broxen** [comment]
Hrm. I'm not getting the same output as you at all. That said, I am able to successfully start the server.
```
$ ./authserver
main
Reading map data from ./data/bin/ folder
 All map data read
 Postprocessing runtime data ..  Hashes filled
```

It almost like QT_NO_DEBUG_STREAM, QT_NO_DEBUG_OUTPUT, and QT_NO_WARNING_OUTPUT have been set, but that's not the case, and I haven't changed any settings.

**Comment by nemerle** [comment]
Will investigate, maybe it's related to ACE debug stuff..

**Comment by nemerle** [comment]
When You have the time, please verify if the logging outputs what it should ?

**Comment by broxen** [comment]
I'm getting barely anything out of output.log and crash.log.

Startup and shutdown events only.

**Comment by nemerle** [comment]
And the log output file ?

**Comment by nemerle** [comment]
Ok, it seems that it's once again the way of redhats :tophat: 
https://bugzilla.redhat.com/show_bug.cgi?id=1227295

**Comment by broxen** [comment]
Ahhh, interesting. I'll play with the local environment settings to see if I can resolve the issue.

I was going to jump on #120... As this is somewhat related, does it provide an argument for switching some of the output from QDebug to something else? This could prevent unexpected behavior due to bad actors at the environment level?

**Comment by nemerle** [comment]
Well, I think I might have a solution for this, could  You try:
add ```#include <QLoggingCategory>``` in AuthServer/main.cpp
add ```QLoggingCategory::setFilterRules("*.debug=true;qt.*.debug=false")``` call before qInstallMessageHandler 
See if this will override the override that overrides the logging ? :smile:

**Comment by nemerle** [comment]
As for #120 I'm almost done with it :)

**Comment by broxen** [comment]
I'm on mobile right now, but I'll test in about 6-7hrs.

RE: #120 You're on fire Nem!

**Comment by broxen** [comment]
Testing QLoggingCategory. This is more output than I've ever gotten from the server, so that seems good. Output.log and Crash.log seem empty

Console Output:
```
$ ./authserver
qt.core.logging: Ignoring malformed logging rule: '*.debug=true;qt.*.debug=false'
Super Entity Game Server
http://segs.sf.net/
Copyright (c) 2006-2017 Super Entity Game Server Team (see Authors.txt)
This software is licensed! (See License.txt for details)

segs v0.4.0 alpha (The Real Beginning)
main
Warning : Reading map data from ./data/bin/ folder
Warning :  All map data read
Warning :  Postprocessing runtime data ..  Hashes filled
(3781|140628839102208) Crypto seed 00000001
(3781|140628830709504) User broxen trying to login from 10.0.0.22.
		id : 64u
		 : succeeded
(3781|140628839102208) Client requesting server list
(3781|140628830709504) Client selected server 1!
(3781|140628822316800) create_cookie still needs a good algorithm.0x00000003
CLIENT VERSION 4
CharacterClient id: 3
CharacterClient slots: 8
(3781|140628918732544) Connection closed
 Telling map server to expect a client with character Sarim,0
(3781|140628813924096) create_cookie still needs a good algorithm.0x0a009d47
1 - 0 - 1
	Sending create entity
	Sending create entity
Received cookie confirm a009d49 - 1
	Sending create entity
	Sending create entity
Received window state 11 - 4
// ...
Received window state 8 - 4
pressed
pressed
(3781|140628839102208) Crypto seed 00000001
(3781|140628830709504) User broxen trying to login from 10.0.0.22.
		id : 64u
		 : succeeded
(3781|140628839102208) Client requesting server list
(3781|140628830709504) Client selected server 1!
(3781|140628822316800) create_cookie still needs a good algorithm.0x00000003
CLIENT VERSION 4
CharacterClient id: 3
CharacterClient slots: 8
(3781|140628918732544) Connection closed
 Telling map server to expect a client with character TestGal,1
(3781|140628813924096) create_cookie still needs a good algorithm.0x0a00cedb
1 - 0 - 1
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
Received cookie confirm a00cedd - 1
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
	Sending create entity
Received window state 11 - 4
// ...
Received window state 8 - 4
pressed
pressed
pressed
pressed
Unhandled command event type 32
Received window state 7 - 2
// ...
Received window state 20 - 4
^CSegmentation fault (core dumped)
```

Output.log
```
Warning : Reading map data from ./data/bin/ folder
Warning :  All map data read
Warning :  Postprocessing runtime data ..  Hashes filled
```

Crash.log
```
 Telling map server to expect a client with character Sarim,1
Sending msg to client 0x7f5ffc019700
^CSegmentation fault (core dumped)
```

**Comment by broxen** [comment]
I think this has been resolved as of #128
